src/agent/mod.rs

```diff
@@ -25,7 +25,34 @@ use wasm_bindgen::JsValue;
 use wasm_bindgen_futures::spawn_local;
 use yew::Callback;
 
+/// Options for the login process
+///
+/// ## Non-exhaustive struct
+///
+/// The struct is "non exhaustive", which means that it is possible to add fields without breaking the API.
+///
+/// In order to create an instance, follow the following patterns:
+///
+/// ```rust
+/// # use reqwest::Url;
+/// # use yew_oauth2::prelude::LoginOptions;
+/// # let url = Url::parse("https://example.com").unwrap();
+/// let opts = LoginOptions {
+///   redirect_url: Some(url),
+///   ..Default::default()
```

Comment:
Functional update syntax doesn't work with non_exhaustive structs, so this won't work (And I'd say it makes constructing options a bit awkward anyway, builder pattern is way nicer). 

At this point, I think that giving those fields `pub(crate)` visibility and thus enforcing builder pattern would lead to less confusing backwards compatible api :)

Reply:
Yea, just figured that out …

> At this point, I think that giving those fields pub(crate) visibility and thus enforcing builder pattern would lead to less confusing backwards compatible api :)

Good point!

Reply:
I am with @kate-shine the builder pattern would be way easier to extend and make it easier to use too.